### ISSUE TITLE
feature request: Support nodejs18.x Lambda runtime

### ISSUE THREAD

**Issue by davecardwell** [localstack/localstack#7209]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

AWS Lambda [now supports](https://docs.aws.amazon.com/lambda/latest/dg/lambda-nodejs.html) the nodejs18.x runtime and it would be great if Localstack could please follow suit.

**Comment by dfangl** [comment]
Hi!
I just opened a PR (linked) for this.
However, this PR will only enable support for our new lambda provider implementation. To use it, (once it is merged), please use `PROVIDER_OVERRIDE_LAMBDA=asf`. It is still not default, but should be (with the end of this day) fairly feature equivalent to the current one.
If you have any issues with this provider, please report here in Github (mentioning using the new one), so we can fix them right away before it becomes default.

It is rather unlikely we will support nodejs18.x in the current provider, if the lambci project is not continued or a fork (currently we use mlupin's fork https://github.com/mLupine/docker-lambda for newer runtimes) releases that image, as we try to establish our new implementation as new standard.

**Comment by davecardwell** [comment]
@dfangl that sounds perfect, thanks for your hard work!

I'll check it out later and will report any issues.

**Comment by Tohaker** [comment]
Is it possible to test this before it's released?

**Comment by dfangl** [comment]
Yes, setting `PROVIDER_OVERRIDE_LAMBDA=asf` on the latest image (please remember to pull!) should already work!

**Comment by Tohaker** [comment]
Hmm, doesn't appear to work still... see this run https://github.com/Carto-Discord/carto/actions/runs/3554458543/jobs/5970601384

I can see it's pulling 1.3.0-dev and I have `PROVIDER_OVERRIDE_LAMBDA=asf` in my docker-compose [file](https://github.com/Carto-Discord/carto/blob/c3a3454fa3a509f3fc2996d91487be9fcf9010a3/packages/integration/docker-compose.yml), is there anything else I'm doing wrong?

**Comment by Tohaker** [comment]
Getting some other strange errors with this override, I'm assuming this is due to the different provider; All requests fail with this in the logs 

```
localstack.utils.container_utils.container_client.ContainerException: Unable to detect IP address for container 8b63803bdf21de31b508765d551f30e7 in network host: Address cannot be empty
```

More logs [here](https://github.com/Carto-Discord/carto/actions/runs/3558992828/jobs/5978077309) (In the `Print docker-compose logs` step)

**Comment by davecardwell** [comment]
@dfangl Unfortunately I haven’t been able to test this for my use case yet.

The initial issue was that arm64 is not supported. I can probably switch to x86_64 in dev as our use of native code is limited, but we’d consider this a _very_ nice to have if not a must have.

The blocker now for our current set up is this error:
```
2022-11-27T20:17:08.035 ERROR --- [   asgi_gw_0] l.aws.handlers.logging     : exception during call chain
Traceback (most recent call last):
  File "/opt/code/localstack/localstack/aws/chain.py", line 90, in handle
    handler(self, self.context, response)
  File "/opt/code/localstack/localstack/aws/handlers/service.py", line 122, in __call__
    handler(chain, context, response)
  File "/opt/code/localstack/localstack/aws/handlers/service.py", line 92, in __call__
    skeleton_response = self.skeleton.invoke(context)
  File "/opt/code/localstack/localstack/aws/skeleton.py", line 153, in invoke
    return self.dispatch_request(context, instance)
  File "/opt/code/localstack/localstack/aws/skeleton.py", line 165, in dispatch_request
    result = handler(context, instance) or {}
  File "/opt/code/localstack/localstack/aws/skeleton.py", line 117, in __call__
    return self.fn(*args, **kwargs)
  File "/opt/code/localstack/localstack/services/awslambda/provider.py", line 154, in create_function
    code=Code(zip_file=code["ZipFile"]),  # TODO: s3
KeyError: 'ZipFile'
2022-11-27T20:17:08.036  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS lambda.CreateFunction => 500 (InternalError)
```

We’re using a basic Terraform `aws_lambda_function` with `s3_bucket = "__local__"`, `s3_key = "/some/path/on/the/host"`. Let me know if you need me to try to create a minimal reproduction.

**Comment by dfangl** [comment]
@Tohaker Please use a bridge network, and not `host` as LAMBDA_DOCKER_NETWORK. Host is currently not compatible and also not recommended.

@davecardwell It seems you are using an old version of the docker image. Please pull the latest image. Arm64 support for native code is on our TODO list, so please stay tuned!

Please also be aware that hot reloading does not yet work, so using the `__local__` key for s3 bucket will not work! Please specify a proper S3 bucket with your code, or use zip file upload for the moment.

**Comment by davecardwell** [comment]
@dfangl Thanks for confirming. Is there an issue I can track for the missing `__local__` support in the new provider?

**Comment by dfangl** [comment]
Yes, you can track it right here: https://github.com/localstack/localstack/issues/7200
Note that we probably have to change the default bucket value of `__local__` to something else, since it does not work with the latest CDK versions due to input validation (even if we would allow it, `__local__` is not a valid S3 bucket name and will be prevented by the client side validation in later versions).

**Comment by dfangl** [comment]
I will now close this issue as support is implemented and tested in the new lambda provider, and probably will not be backported to the old one.
